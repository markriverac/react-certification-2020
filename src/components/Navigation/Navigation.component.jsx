import React, { useContext, useState } from 'react';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import './Navigation.styles.css';
import { Link } from 'react-router-dom';
import UserContext from '../../providers/UserProvider/User.context';

function Navigation() {
  const userContext = useContext(UserContext);
  const [username, setusername] = useState('');
  const { userState, logIn } = useContext(UserContext);

  function authenticate(event) {
    event.preventDefault();
    logIn(username);
    console.log(userContext);
  }

  function onUsernameChange(event) {
    setusername(event.target.value);
  }

  const loginForm = (
    <form className="form-inline" onSubmit={authenticate}>
      <input
        className="form-control mr-sm-2 login-input"
        type="search"
        placeholder="username"
        aria-label="Search"
        value={username}
        onChange={onUsernameChange}
      />
      <button className="btn btn-outline-secondary my-2 my-sm-0" type="submit">
        LogIn
      </button>
    </form>
  );

  const welcome = <Navbar.Brand>Welcome {userState.name}</Navbar.Brand>;
  return (
    <Navbar bg="dark" variant="dark" expand="lg" sticky="top">
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Brand href="#home">Find Your video</Navbar.Brand>
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="mr-auto">
          <Link to="/" className="nav-link">
            Home
          </Link>
          {userState.isAuth && (
            <Link to="/favorites" className="nav-link">
              Favorites
            </Link>
          )}
        </Nav>
      </Navbar.Collapse>
      {!userState.isAuth ? loginForm : welcome}
    </Navbar>
  );
}

export default Navigation;
