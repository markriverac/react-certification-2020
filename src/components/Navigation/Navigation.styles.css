.login-input {
  background-color: transparent !important;
  border: none !important;
  border-bottom: solid gray 1px !important;
}
