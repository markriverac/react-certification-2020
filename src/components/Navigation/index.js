/* istanbul ignore file */
export { default } from './Navigation.component';
