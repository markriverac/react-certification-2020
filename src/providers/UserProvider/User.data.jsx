export const userData = { isAuth: false, name: '', favorites: [] };
