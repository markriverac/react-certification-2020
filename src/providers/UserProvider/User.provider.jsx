import React, { useState } from 'react';
import UserContext from './User.context';
import { userData } from './User.data';

function UserProvider({ children }) {
  const [userState, setuserState] = useState(userData);

  function logIn(username) {
    setuserState({ ...userState, name: username, isAuth: true });
  }
  function logOut() {
    setuserState({ ...userState, isAuth: false });
  }
  function addFavorite(video) {
    const newFavorites = userState.favorites;
    newFavorites.push(video);
    setuserState({ ...userState, favorites: newFavorites });
  }
  function removeFavorite(video) {
    const favoriteVideos = userState.favorites;
    const videoIndex = favoriteVideos.indexOf(video);
    if (videoIndex > -1) {
      favoriteVideos.splice(videoIndex, 1);
      setuserState({ ...userState, favorites: favoriteVideos });
    }
  }

  return (
    <UserContext.Provider
      value={{ userState, logIn, logOut, addFavorite, removeFavorite }}
    >
      {children}
    </UserContext.Provider>
  );
}

export default UserProvider;
