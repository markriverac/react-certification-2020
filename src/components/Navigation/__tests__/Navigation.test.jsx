import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';
import UserProvider from '../../../providers/UserProvider/User.provider';
import Navigation from '../Navigation.component';
import UserContext from '../../../providers/UserProvider/User.context';

Enzyme.configure({ adapter: new Adapter() });

it('Renders button without exploding', () => {
  const navigation = renderer.create(
    <BrowserRouter>
      <UserProvider>
        <Navigation />
      </UserProvider>
    </BrowserRouter>
  );
  const tree = navigation.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders login form if not in', () => {
  const navigation = renderer.create(
    <BrowserRouter>
      <UserProvider>
        <Navigation />
      </UserProvider>
    </BrowserRouter>
  );
  const tree = navigation.toJSON();
  expect(tree.children[3].type).toEqual('form');
});

it('renders username when logged in', () => {
  const userState = { isAuth: true };
  const navigation = renderer.create(
    <BrowserRouter>
      <UserContext.Provider value={{ userState }}>
        <Navigation />
      </UserContext.Provider>
    </BrowserRouter>
  );
  const tree = navigation.toJSON();
  expect(tree.children[3].type).toEqual('span');
});
