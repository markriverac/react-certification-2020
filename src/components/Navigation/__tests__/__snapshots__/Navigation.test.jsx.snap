// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders button without exploding 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark sticky-top"
>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <a
    className="navbar-brand"
    href="#home"
  >
    Find Your video
  </a>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
  </div>
  <form
    className="form-inline"
    onSubmit={[Function]}
  >
    <input
      aria-label="Search"
      className="form-control mr-sm-2 login-input"
      onChange={[Function]}
      placeholder="username"
      type="search"
      value=""
    />
    <button
      className="btn btn-outline-secondary my-2 my-sm-0"
      type="submit"
    >
      LogIn
    </button>
  </form>
</nav>
`;
